tbtadm/controller.cpp

```diff
@@ -649,20 +654,23 @@ void tbtadm::Controller::approve(const fs::path& dir) try
             return dist(eng);
         });
 
-        File key(dir / keyFilename, File::Mode::Write);
+        std::ofstream key((dir / keyFilename).string());
         key << keyStream.str();
     }
 
-    authorized = File(dir / authorizedFilename, File::Mode::Write);
+    authorized.clear();
     authorized << 1;
 
     m_out << "Authorized\n";
     if (m_sl == SECURITY_LEVEL_SECURE && !m_once)
     {
-        File keyACL(acltree / readAndTrim(dir / uniqueIDFilename) / keyFilename,
-                    File::Mode::Write,
-                    O_CREAT,
-                    S_IRUSR);
+        auto fn = acltree / readAndTrim(dir / uniqueIDFilename) / keyFilename;
+
+        /* Open with Linux system call since C++ cannot handle it */
+        int fd = open(fn.string().c_str(), O_WRONLY | O_CREAT, S_IRUSR);
+        close(fd);
```

Comment:
So you are effectively creating a file here? I'm sure you can pass the descriptor to ofstream and friends without this kind of hackery. Furthermore, I'm pretty sure C++ standard library supports creating a file :-)

Reply:
I am creating file with needed permissions, AFAIK C++ does not allow this, @ybernat could you confirm this?

Reply:
Still you should be able to pass that fd directly to ofstream, I think.

Reply:
Unfortunately, there is no translation between neither `fd` nor `FILE*` to `fstream` facilities (unlike the translation existing between `fd` and `FILE*`), probably due to the buffering happens in `iostream` level (but this is just my wild guess).

For setting permissions (after creating the file, which can be done with `fstream`, of course), you can use [`filesystem::permissions()`](http://en.cppreference.com/w/cpp/filesystem/permissions) for setting the desired ones (or maybe it's better to set `umask` in first place?)

Reply:
Looks like setting umask() is the easiest way.